src/riot-rs-rt/Cargo.toml

```diff
@@ -44,8 +44,14 @@ debug-console = ["riot-rs-debug/debug-console"]
 executor-single-thread = []
 silent-panic = []
 _panic-handler = []
-critical-section-single-core = ["cortex-m?/critical-section-single-core"]
-multicore = ["embassy-rp?/critical-section-impl"]
+critical-section-single-core = [
+  "cortex-m?/critical-section-single-core",
+  "portable-atomic/unsafe-assume-single-core",
+]
+multicore = [
+  "embassy-rp?/critical-section-impl",
+  "portable-atomic/critical-section",
+]
```

Comment:
cc @ROMemories  as the author of #363. 

Any reason why the  `critical-section` feature instead of `unsafe-assume-single-core` should be picked if the RP2040 is used as single-core?
 If I understand the [docs](https://docs.rs/portable-atomic/latest/portable_atomic/#optional-features-unsafe-assume-single-core) correctly, it has better performance and is safe as long as it's single-core and runs in privileged mode (which RIOT-rs always does).

Reply:
Hmm, so turns out this is unfortunately not possible, because `unsafe_assume-single-core` doesn't work on platforms that do support CAS. So we can't select it here through the feature. and we obviously can't enable it just for the `rp2040 context`, because then it fails on multicore.
Any idea how to solve this?
I dropped the commit for now.